torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -556,3 +584,35 @@ def forward(ctx, param_group: FSDPParamGroup, *inputs: torch.Tensor):
     def backward(ctx, *grads: torch.Tensor):
         ctx.param_group.post_backward()
         return (None,) + grads
+
```

Comment:
For context, the existing multi-grad hook does not work so well with the existing `post_backward` because the last parameter's `.grad` is not assigned yet when the mulit-grad hook runs (expected behavior).

We actually prefer this mulit-post-accumulate-grad hook, where the last parameter's `.grad` has already been assigned.